src/instructlab/lab.py

```diff
@@ -685,16 +716,32 @@ def chat(
 def download(ctx, repository, release, filename, model_dir):
     """Download the model(s) to train"""
     click.echo(f"Downloading model from {repository}@{release} to {model_dir}...")
+    if "HF_TOKEN" not in os.environ and repository != "instructlab/merlinite-7b-lab-GGUF":
+        raise ValueError("HF_TOKEN var needs to be set in your environment to download HF Model")
     try:
         if ctx.obj is not None:
             hf_logging.set_verbosity(ctx.obj.config.general.log_level.upper())
-
-        hf_hub_download(
-            repo_id=repository,
-            revision=release,
-            filename=filename,
-            local_dir=model_dir,
-        )
+        files = huggingface_hub.list_repo_files(repo_id=repository, token=os.getenv("HF_TOKEN"))
+        if any(".safetensors" in string for string in files):
+            if not os.path.exists(os.path.join(model_dir, "safetensors")):
+                os.mkdir(os.path.join(model_dir, "safetensors"))
+            for f in files:
+                # if fullmodel we need to download all safetensors
+                hf_hub_download(
+                    token=os.getenv("HF_TOKEN"),
+                    repo_id=repository,
+                    revision=release,
+                    filename=f,
+                    local_dir=os.path.join(model_dir, "safetensors"),
```

Comment:
we will need to download more than just the safetensor files. we will also need the config.json for the model's metadata as well as all the tokenizer files

It would be better to download the whole repo IMO

Reply:
Sorry, I just named the dir safetenors. It downloads all the files in the repo

Reply:
moved to another branch